### ISSUE TITLE
OpenID will be deprecated

### ISSUE THREAD

**Issue by FelixSFD** [SOBotics/SwiftChatSE#8]
https://meta.stackexchange.com/q/307647/347985

It looks like the library [is using OpenID](https://github.com/SOBotics/SwiftChatSE/blob/master/Sources/SwiftChatSE/Client.swift#L507). We need to replace it with the e-mail-based authentication.

If we need help, [Tim Post](https://meta.stackexchange.com/a/307649/347985) is inviting us to contact SE.

**Comment by NobodyNada** [comment]
Hmm, thanks!  Our bots don’t “directly” use OpenID, but (as of a few years ago, when the bot frameworks were created), `openid.stackexchange.com` is the “login processor” that all the SE sites’ login forms use.  (I made SwiftChatSE’s login processor by studying the requests made by the web browser when logging in and duplicating them in code.)

If OpenID is shut down, then it seems likely the login processor will completely change, which means all bots need to be updated to the new process.  We can reverse-engineer the login process again, or it might be worth contacting Tim.

**Comment by NobodyNada** [comment]
We’ll also probably want to let the Charcoal folks know about this too, along with anyone else who runs bots.

**Comment by NobodyNada** [comment]
On a related note, there is almost zero documentation for the login and chat APIs; anyone who wants to create a new bot library has to poke around the code for various libraries and bots, find obscure meta posts, and experiment with requests until it works.  I think it would be worth creating a page on sobotics.org, as part of this project, with extensive documentation on login and chat.

**Comment by Bhargav-Rao** [comment]
> On a related note, there is almost zero documentation for the login and chat APIs; anyone who wants to create a new bot library has to poke around the code for various libraries and bots, find obscure meta posts, and experiment with requests until it works. I think it would be worth creating a page on sobotics.org, as part of this project, with extensive documentation on login and chat.

Yes, this. Given that we have migrated over from GH Pages to Wordpress for the main site, the GH Pages repo is now empty. I'm planning to convert it into our blog. I don't know enough Jekyll to do that, but we can come up with something. For the moment, let's utilize SO Channels/Teams for posting questions regarding it, where we can all edit the answer. Once SO Channels moves out of beta, they'll kick us out as well, after which we can put that content on the blog and make it public to everyone.

**Comment by Zomis** [comment]
I'm also interested in a solution to this as I found that my code also refers to openid.stackexchange.com here: https://github.com/Zomis/Duga/blob/develop/duga-core/src/main/java/net/zomis/duga/chat/StackExchangeLogin.java#L70

I also created a chat room on Stack Exchange for this: https://chat.stackexchange.com/rooms/74338/chat-bot-migration-from-openid

**Comment by Bhargav-Rao** [comment]
@Zomis Another one of our [Java chat libraries use OpenID.](https://github.com/Tunaki/chatexchange/issues/6). Do let us know if you manage to find a way to port to a different login. 

Jeremy Banks is trying to port his python library Chat.X, and has made some headway using StackAuth. 

I'm sure that there would be a way to fix it. Tim Post is quite aware of the issue (after I ranted about it in the TL)

**Comment by FelixSFD** [comment]
I've updated the Java-lib: https://github.com/Tunaki/chatexchange/pull/7

We basically need to fetch the `fkey` from the login-form on `{host}/users/login` and `POST` it along with the credentials to the same page. Then grab the cookies and use them on `chat.{host}`. It's actually much easier than using OpenID :-D
